### DISCUSSION TITLE
Kind attention, any help (`contrib/check-config.sh`)

### DISCUSSION THREAD

**Discussion by Universalizer** [moby/moby#47033] [Q&A] [answered]
https://github.com/moby/moby/blob/master/contrib/check-config.sh

Results following :-

`> .wrap_bad "  (RHEL7/CentOS7" "User namespaces disabled; add 'user_namespace.enable=1' to boot command line)"
bash: syntax error near unexpected token `('`

`[Process completed (code 2) - press Enter]`
<hr>
I have done a search for similar discussions* for this,

bash: syntax error near unexpected token `('

And

https://github.com/search?q=repo%3Amoby%2Fmoby+%3E+.wrap_bad+%22++%28RHEL7%2FCentOS7%22+%22User+namespaces+disabled%3B+add+%27user_namespace.enable%3D1%27+to+boot+command+line%29%22+bash%3A+syntax+error+near+unexpected+token+%60%28%27++%5BProcess+completed+%28code+2%29+-+press+Enter%5D&type=issues

**Comment by AkihiroSuda** [comment]
What's your distro? What is your `/bin/sh` implementation?

You may have to run the script with `bash ./check-config.sh`

**Reply by thaJeztah** [reply]
Oh! Is was looking at this Yesterday, but didn't comment; I wonder if it is because `wrap_bad` expects 2 arguments; https://github.com/moby/moby/blob/c41f94eda3dee362e3ed8a0ad8fe62d7d0bc68b2/contrib/check-config.sh#L88-L90

But it looks like this line is only passing one; https://github.com/moby/moby/blob/c41f94eda3dee362e3ed8a0ad8fe62d7d0bc68b2/contrib/check-config.sh#L142-L143

**Reply by thaJeztah** [reply]
Oh! Ignore me; they are 2 separate strings (`"  (RHEL7/CentOS7"` is a separate string), so that should be fine

**Comment by Universalizer** [comment]
Android Termux https://github.com/termux/termux-app

**Reply by AkihiroSuda** [reply]
Android is not supported
- https://github.com/moby/moby/issues/41111

**Comment by Universalizer** [comment, marked as answer]
This means `Docker supports only Rooted Android`.
Thanks, Closing this discussion.

**Comment by Universalizer** [comment]
`Just information`

An open-source, next-generation "runc" that empowers rootless containers to run workloads such as Systemd, Docker, Kubernetes, just like VMs.

https://github.com/nestybox/sysbox#introduction
